npm/ipsm/ipsm.go

```diff
@@ -613,57 +605,59 @@ func (ipsMgr *IpsetManager) DestroyNpmIpsets() error {
 		return nil
 	}
 
-	entry := &ipsEntry{
-		operationFlag: util.IpsetFlushFlag,
-	}
-
+	destroyFailureCount := 0
 	for _, ipsetName := range ipsetLists {
-		entry := &ipsEntry{
+		flushEntry := &ipsEntry{
 			operationFlag: util.IpsetFlushFlag,
 			set:           ipsetName,
 		}
+		_, flushError := ipsMgr.run(flushEntry)
 
-		if _, err := ipsMgr.run(entry); err != nil {
-			metrics.SendErrorLogAndMetric(util.IpsmID, "{DestroyNpmIpsets} Error: failed to flush ipset %s", ipsetName)
+		deleteEntry := &ipsEntry{
+			operationFlag: util.IpsetDestroyFlag,
+			set:           ipsetName,
 		}
-	}
+		_, destroyError := ipsMgr.run(deleteEntry)
 
-	for _, ipsetName := range ipsetLists {
-		entry.operationFlag = util.IpsetDestroyFlag
-		entry.set = ipsetName
-		if _, err := ipsMgr.run(entry); err != nil {
+		if flushError != nil {
+			metrics.SendErrorLogAndMetric(util.IpsmID, "{DestroyNpmIpsets} Error: failed to flush ipset %s", ipsetName)
+		}
+		if destroyError != nil {
+			destroyFailureCount++
 			metrics.SendErrorLogAndMetric(util.IpsmID, "{DestroyNpmIpsets} Error: failed to destroy ipset %s", ipsetName)
 		}
```

Comment:
may use `continue` here and remove below `if` condition?

Reply:
I think all the if statements here are independent since if you fail to flush you technically can still succeed on destroy